src/main/java/uk/gov/hmcts/reform/professionalapi/domain/entities/Organisation.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.professionalapi.domain.entities;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Entity(name = "organisation")
+@Getter
+@NoArgsConstructor
+public class Organisation {
+
+    @Id
+    private String id;
```

Comment:
I think it'd be worth having an abstract superclass for our entities that handles the UUID IDs and datestamps for us, kind of like:
```java
package uk.gov.hmcts.reform.refdata.domain;

import java.time.LocalDateTime;
import java.util.UUID;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;

import lombok.Data;

/**
 * @author Tabby Cromarty
 */
@MappedSuperclass
@Data
public abstract class AbstractEntity {

  @Id
  @GeneratedValue(generator = "system-uuid")
  @GenericGenerator(name = "system-uuid", strategy = "uuid")
  private UUID id;
  @CreationTimestamp
  LocalDateTime dateCreated;
  @UpdateTimestamp
  LocalDateTime lastUpdated;

}
```

Also this would mean we can use the native UUID datatype in the database because there is no way in hell we want our primary key columns to be strings :S

The actual entitities can then look like
```java
package uk.gov.hmcts.reform.refdata.domain;

import javax.persistence.Entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Tabby Cromarty
 */
@Entity
@Data
@EqualsAndHashCode(callSuper = true)
public class EmailDomain extends AbstractEntity {

  String host;

}
```

Reply:
Yes I think we should use a UUID datatype since that models the database type UUID better.  Not too fussed if its in an Abstract class or not.